### PR TITLE
GV-10. adding Edge type

### PR DESCRIPTION
**Pull request by Sveta-1999**
This PR closes [GV-10](https://graphviz.atlassian.net/browse/GV-10?atlOrigin=eyJpIjoiNzQzMjA5YTQ1OTY5NGEyMGI1ZjBjYTAyMzAzODQ1ZjgiLCJwIjoiaiJ9) ticket.
Added methods getter, setter which is the initialization of member data's.

### CODE REVIEW

File: examples/golden_graph/edge.hpp
Comment (aramhovakimyan) [inline examples/golden_graph/edge.hpp:1]: Lets change header guards name like this:
EDGE_CLASS -> EXAMPLES_GOLDEN_GRAPH_EDGE_HPP

File: examples/golden_graph/edge.hpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#ifndef EXAMPLE_GOLDEN_GRAPH_EDGE_HPP
+#define EXAMOLE_GOLDEN_GRAPH_EDGE_HPP
+
+class Edge {
+public:
+    Edge(int id_1, int id_2) : _id_1(id_1), _id_2(id_2) {}
```
Comment (SarkisKhachatryan) [inline examples/golden_graph/edge.hpp]: move to cpp

File: examples/golden_graph/edge.hpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public:
+    void set_id_1(int);
```
Comment (SarkisKhachatryan) [inline examples/golden_graph/edge.hpp]: add argument names

File: examples/golden_graph/edge.cpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#include "edge.hpp"
+
+void Edge::set_id_1(int id_1) {
+    _id_1 = id_1;
+
```
Comment (SarkisKhachatryan) [inline examples/golden_graph/edge.cpp:4]: remove redundant line

**Review by SarkisKhachatryan — CHANGES_REQUESTED** [review]

**Review by SarkisKhachatryan — DISMISSED** [review]

**Review by SarkisKhachatryan — APPROVED** [review]

**Pull request merged**